packages/lab/src/components/Flow/hooks/useFlowNode.ts

```diff
@@ -1,24 +1,53 @@
-import { Edge, Node, useEdges, useNodes } from "reactflow";
+import { useMemo } from "react";
+import { Node, Edge, ReactFlowState, useStore } from "reactflow";
 
-export function useFlowNode<T = any>(id: string) {
-  const nodes = useNodes<T>();
-  const edges = useEdges();
+export function useFlowNode<T extends Node = Node>(
+  id: string
+): Node | undefined {
+  const nodeSelector = useMemo(
+    () => (state: ReactFlowState) =>
+      state.getNodes().find((n: Node) => n.id === id) as T | undefined,
+    [id]
+  );
+  return useStore<T | undefined>(nodeSelector);
```

Comment:
These types are also already inferred. also, `useCallback`
same for the other hooks
```suggestion
  const nodeSelector = useCallback(
    (state: ReactFlowState) =>
      state.getNodes().find((n): n is T => n.id === id),
    [id]
  );
  return useStore(nodeSelector);
```